cognite_toolkit/_cdf_tk/utils/fileio/_compression.py

```diff
@@ -0,0 +1,78 @@
+import gzip
+from abc import ABC, abstractmethod
+from collections.abc import Mapping
+from io import TextIOWrapper
+from pathlib import Path
+from typing import ClassVar, Literal
+
+from cognite_toolkit._cdf_tk.exceptions import ToolkitValueError
+from cognite_toolkit._cdf_tk.utils import humanize_collection
+from cognite_toolkit._cdf_tk.utils._auxiliary import get_concrete_subclasses
+
+
+class Compression(ABC):
+    encoding = "utf-8"
+    newline = "\n"
+    name: ClassVar[str]
+    file_suffix: ClassVar[str]
+
+    def __init__(self, filepath: Path) -> None:
+        self.filepath = filepath
+
+    @abstractmethod
+    def open(self, mode: Literal["r", "w"]) -> TextIOWrapper:
+        """Open the compressed file and return a file-like object."""
+        raise NotImplementedError("This method should be implemented in subclasses.")
+
+    @classmethod
+    def from_filepath(cls, filepath: Path) -> "Compression":
+        return COMPRESSION_BY_SUFFIX.get(filepath.suffix, NoneCompression)(filepath=filepath)
```

Comment:
I wonder if it would be safer to raise an error if the suffix is unknown instead of falling back to None.
If someone with a new version of toolkit compresses with a new format, and someone with an old toolkit version tries to decrompress it, they will probably get some weird errors.

Reply:
But how do you deal with non-compressed files? For uncompressed files, you will just get an error if you do what you suggest. 

And you will not get a wired error message, just that the file format x is not supported